domain/src/main/java/org/oppia/android/domain/exploration/ExplorationProgressController.kt

```diff
@@ -418,13 +472,70 @@ class ExplorationProgressController @Inject constructor(
             AsyncResult.failed<EphemeralState>(e)
           }
         }
-        ExplorationProgress.PlayStage.VIEWING_STATE ->
+        ExplorationProgress.PlayStage.VIEWING_STATE -> {
+          asyncDataSubscriptionManager.notifyChangeAsync(EXPLORATION_CHECKPOINT_DATA_PROVIDER_ID)
```

Comment:
One thing I am stuck on is that the `DataProvider` should not be re-queried unless we notify that the `DataProvider` has changed using this line. `asyncDataSubscriptionManager.notifyChangeAsync(EXPLORATION_CHECKPOINT_DATA_PROVIDER_ID)`. 
But even after removing this line, the `DataProvider` is getting re-queried 2-3 times.